src/components/inputs/Radio.tsx

```diff
@@ -99,78 +169,106 @@ const Radio = React.forwardRef<HTMLDivElement, RadioProps>(function RadioFn(
 		(e) => {
 			if (!disabled) {
-				if (uncontrolledMode && !e.isDefaultPrevented()) {
-					setIsActive((prevState) => !prevState);
-				}
-				if (onClick) {
-					onClick(e);
+				if (uncontrolledMode && !e.defaultPrevented) {
+					setIsChecked((prevState) => !prevState);
 				}
+				onClick?.(e);
 			}
 		},
 		[disabled, onClick, uncontrolledMode]
 	);
 
-	const keyEvents = useMemo(() => getKeyboardPreset('button', handleOnClick), [handleOnClick]);
-	useKeyboard(radioRef, keyEvents);
+	const onChangeHandler = useCallback<React.ChangeEventHandler<HTMLInputElement>>(() => {
+		// do nothing
+		// TODO: CDS-174: update state here and not in the click. In controlled mode, do not update the internal state,
+		// 	just call the external onChange
+	}, []);
 
 	useEffect(() => {
-		onChange && onChange(isActive);
-	}, [onChange, isActive]);
+		// TODO: CDS-174: remove this effect, call onChange only from the handler
+		onChange?.(isChecked);
+	}, [onChange, isChecked]);
 
 	useEffect(() => {
-		typeof checked !== 'undefined' && setIsActive(checked);
+		if (checked !== undefined) {
+			setIsChecked(checked);
+		}
 	}, [checked]);
 
+	const labelWithClick = useMemo(
+		() =>
+			(typeof label === 'string' && (
+				<Label disabled={disabled} size={RADIO_SIZE[size].label} htmlFor={id} ref={labelRef}>
+					{label}
+				</Label>
+			)) ||
+			(React.isValidElement<{ onClick?: React.MouseEventHandler }>(label) &&
+				React.cloneElement(label, {
+					onClick: (event: React.MouseEvent) => {
+						label.props?.onClick?.(event);
+						if (!event.defaultPrevented && radioInputRef.current) {
+							radioInputRef.current.click();
+						}
+					}
+				})) ||
+			label,
+		[disabled, id, label, radioInputRef, size]
+	);
+
 	return (
 		<RadioContainer
-			ref={radioRef}
+			ref={containerRef}
 			width="100%"
 			height="auto"
 			mainAlignment="flex-start"
 			crossAlignment="center"
 			orientation="horizontal"
 			padding={padding}
-			style={{ cursor: disabled ? 'default' : 'pointer' }}
-			onClick={handleOnClick}
 			$disabled={disabled}
-			tabIndex={disabled ? -1 : 0}
 			$iconColor={iconColor}
-			{...rest}
+			gap={'0.5rem'}
 		>
-			<StyledIcon
-				size={size === 'medium' ? 'large' : 'medium'}
-				checked={isActive}
+			<RadioInput
+				type="radio"
+				{...rest}
+				id={id}
+				ref={radioInputRef}
+				checked={isChecked}
+				onClick={onClickHandler}
+				onChange={onChangeHandler}
 				disabled={disabled}
-				icon="RadioButtonOn"
-				color={iconColor}
+				$color={iconColor}
+				$size={RADIO_SIZE[size].icon}
 			/>
-			{typeof label === 'string' ? (
-				<CustomText
-					disabled={disabled}
-					size={size === 'medium' ? 'medium' : 'small'}
-					color="gray0"
-					weight="regular"
-				>
-					{label}
-				</CustomText>
-			) : (
-				label
-			)}
+			{labelWithClick}
 		</RadioContainer>
 	);
 });
 
-export { Radio, RadioProps };
+const Radio = RadioComponent as RadioType;
+
+export { RadioComponent, Radio, RadioProps };
```

Comment:
Why do you export both Radio and RadioComponent?

Reply:
I needed to export both the components here because of the styleguide, which analyze the file, search for a component structure and creates the page for that component. Without exporting it, the doc page is not composed well.
In the index file which is in charge of exporting the components of the library for the npm package, I specified only Radio component though, so that the RadioComponent is only exported internally.